src/python/pants/util/contextutil.py

```diff
@@ -170,11 +177,14 @@ def open_zip(path_or_file, *args, **kwargs):
 
     :API: public
   """
+  if not path_or_file:
+    raise InvalidZipPath('Invalid zip location: {}'.format(path_or_file))
+  allowZip64 = kwargs.pop('allowZip64', True)
   try:
-    allowZip64 = kwargs.pop('allowZip64', True)
     zf = zipfile.ZipFile(path_or_file, *args, allowZip64=allowZip64, **kwargs)
   except zipfile.BadZipfile as bze:
-    raise zipfile.BadZipfile("Bad Zipfile {0}: {1}".format(path_or_file, bze))
+    # Print the realpath in order to follow symlinks back to the problem source file.
+    raise zipfile.BadZipfile("Bad Zipfile {0}: {1}".format(os.path.realpath(path_or_file), bze))
```

Comment:
Maybe, in a new PR, a utility to describe a path could be added and used. In this context, there is no notion of symlink, but realpath makes some sense - ie, lets report the fully resolved terminator. The utility could, if the path is a symlink, recursively follow and build a lits of all paths terminating in the realpath - handing back this whole thing as a string for use in exceptions or otherwise.